### PR TITLE
docs: Add 1.12 upgrade note for pending removal builtins

### PR DESCRIPTION
**Pull request by mpalmi**
Addresses VAULT-11825.

### CODE REVIEW

File: website/content/docs/upgrading/upgrade-to-1.12.x.mdx [hunk 1]
```diff
@@ -93,7 +93,11 @@ marked with the `Pending Removal` status. Any attempt to unseal Vault with
 mounts backed by one of these builtin plugins will result in an immediate
 shutdown of the Vault core.
 
-It is recommended that set the
+-> **NOTE** In the event that an external plugin with the same name and type as
+a deprecated builtin is deregistered, any subsequent unseal of Vault will also
```
Comment (mladlow) [inline website/content/docs/upgrading/upgrade-to-1.12.x.mdx:97]: Could you add an example? Like: 
```suggestion
a deprecated builtin is deregistered (ex: a plugin named "blah" with type "standaloneblah"), any subsequent unseal of Vault will also
```

**Review by mladlow — APPROVED** [review]
I would like an example but approving anyway.

**Comment by mladlow** [comment]
I am almost certain that this needs to be backported to 1.12 to show up on the website now, unless something changed that I missed. I added the label and will approve the backport though.

**Comment by mpalmi** [comment]
> I am almost certain that this needs to be backported to 1.12 to show up on the website now, unless something changed that I missed. I added the label and will approve the backport though.

Thanks! For some reason I thought we didn't backport upgrade notes, but I probably just misunderstood.

**Pull request merged**